### PR TITLE
Ag procwait bugfix

### PR DESCRIPTION
**Pull request by fredex42**
ensuring that the responder waits properly for the called process and reaps it. minor logging improvement.

### CODE REVIEW

**Comment by fredex42** [comment]
@NickPapacostas could you review if you've got a sec? thanks

File: CDS/cdsresponder/cdsresponder.rb [hunk 1]
```diff
@@ -187,8 +188,13 @@ def ThreadFunc
 
           triggerfile=OutputTriggerFile(trigger_content, msg.id)
 
-          @pid = spawn("cds_run", "--route", @routefile, "--#{@cdsarg}", triggerfile)
-          msg=FinishedNotification.new(@routename, $?.exitstatus, GetLogfile(msg.id))
+          cmdline = "cds_run --route \"#{@routefile}\" #{@cdsarg} #{triggerfile}"
```
Comment (NickPapacostas) [inline CDS/cdsresponder/cdsresponder.rb]: did we lose the "--" in front of cdsarg?

File: CDS/cdsresponder/cdsresponder.rb [hunk 2]
```diff
@@ -187,8 +188,13 @@ def ThreadFunc
...
+          @logger.debug("Commandline is #{cmdline}")
+          @pid = spawn(cmdline)
+
+          exitstatus = Process.wait @pid
```
Comment (NickPapacostas) [inline CDS/cdsresponder/cdsresponder.rb]: not sure what the consequences of making this synchronous are... as long as you're aware i'm cool with it ^_^

File: CDS/cdsresponder/cdsresponder.rb (hunk 1 above)
Comment (fredex42) [inline CDS/cdsresponder/cdsresponder.rb]: yeh, the -- is already in the var

File: CDS/cdsresponder/cdsresponder.rb (hunk 2 above)
Comment (fredex42) [inline CDS/cdsresponder/cdsresponder.rb]: yep that's how it's meant to be :)

**Pull request merged**